### PR TITLE
getBlockByHeight and getBlockByPosition functions

### PR DESCRIPTION
**Pull request by ramirez915**
both functions pass their corresponding tests

### CODE REVIEW

**Comment by ramirez915** [comment]
Just to make sure, just for the AddBlock for now? Or should also for the other functions?

**Comment by kstola2** [comment]
All functions would be best. Also, change the opening file to assume that the file is already created, so no `create` flag.

**Comment by ramirez915** [comment]
Cool, will work on that. Quick thing, there's a log.Fatalln(), should I leave it as is or replace it with the errors.New()?

**Comment by kstola2** [comment]
Replace it with an error

**Comment by ramirez915** [comment]
Alright

File: producer/src/blockchain/blockchain.go [hunk 1]
```diff
@@ -48,27 +49,30 @@ func AddBlock(b block.Block, filename string, databaseName string) error { // Ad
...
 	}
 
 	database, err := sql.Open("sqlite3", databaseName)
 	// Checks if the opening was successful
 	if err != nil {
-		return err
+		return errors.New("Unable to open sqlite3 database")
 	}
 
+	defer file.Close()
```
Comment (ramirez915) [inline producer/src/blockchain/blockchain.go]: I think closing the file was forgotten so I added it in

**Comment by ramirez915** [comment]
I have made the changes to the three functions

**Comment by kstola2** [comment]
Looks like those particular functions are working, now we just need `GetBlockByHash` and the final test.

**Comment by ramirez915** [comment]
getBlockByHash is completed and it passes it's corresponding test. I forgot to merge to dev prior to these changes though

**Pull request merged**